gnomad_qc/v3/sample_qc/sample_qc.py

```diff
@@ -424,16 +426,44 @@ def main(args):
     if args.impute_sex:
         compute_sex().write(sex.path, overwrite=args.overwrite)
     elif args.reannotate_sex:
-        sex_ht = sex.ht().checkpoint('gs://gnomad-tmp/sex_ht_checkpoint.ht', overwrite=True)  # Copy HT to temp location to overwrite annotation
-        x_ploidy_cutoff, y_ploidy_cutoff = get_ploidy_cutoffs(sex_ht, f_stat_cutoff=0.5)
+        # Copy HT to temp location to overwrite annotation
+        sex_ht = sex.ht().checkpoint('gs://gnomad-tmp/sex_ht_checkpoint.ht', overwrite=True)
+        hard_filter_ht = compute_hard_filters(args.min_cov, include_sex_filter=False)
+        # Copy HT to temp location because it uses sex_ht for chr20 coverage
+        hard_filter_ht = hard_filter_ht.checkpoint('gs://gnomad-tmp/hardfilter_checkpoint.ht', overwrite=True)
+        x_ploidy_cutoff, y_ploidy_cutoff = get_ploidy_cutoffs(
+            sex_ht.filter(hl.is_missing(hard_filter_ht[sex_ht.key])),
+            f_stat_cutoff=0.5
+        )
+        upper_x = args.upper_x if args.upper_x else x_ploidy_cutoff[0]
+        lower_xx = args.lower_xx if args.lower_xx else x_ploidy_cutoff[1][0]
+        upper_xx = args.upper_xx if args.upper_xx else x_ploidy_cutoff[1][1]
+        lower_xxx = args.lower_xxx if args.lower_xxx else x_ploidy_cutoff[2]
+        lower_y = args.lower_y if args.lower_y else y_ploidy_cutoff[0][0]
+        upper_y = args.upper_y if args.upper_y else y_ploidy_cutoff[0][1]
+        lower_yy = args.lower_yy if args.lower_yy else y_ploidy_cutoff[1]
+
```

Comment:
This may be more of a style thing but since these cut offs are used three times, an alternative approach could be to create the struct we use to annotate the globals here and then reference it in the rest of  this section
```suggestion
x_ploidy_cutoffs = hl.struct(
    upper_x = args.upper_x if args.upper_x else x_ploidy_cutoff[0],
    lower_xx = args.lower_xx if args.lower_xx else x_ploidy_cutoff[1][0],
    upper_xx = args.upper_xx if args.upper_xx else x_ploidy_cutoff[1][1],
    lower_xxx = args.lower_xxx if args.lower_xxx else x_ploidy_cutoff[2]
)
y_ploidy_cutoffs=hl.struct(
    lower_y = args.lower_y if args.lower_y else y_ploidy_cutoff[0][0],
    upper_y = args.upper_y if args.upper_y else y_ploidy_cutoff[0][1],
    lower_yy = args.lower_yy if args.lower_yy else y_ploidy_cutoff[1]
)